packages/composables/src/composables/useCart/index.ts

```diff
@@ -4,57 +4,75 @@ import {
 
-const params: UseCartFactoryParams<Cart, CartItem, Product, Coupon> = {
+// Return types from Vendure mutations on Cart (Order) are unions of Order and something else (i.e. OrderModificationError).
+// When we return this union type, TypeScript complains that there are no properties in common with order as it returns only __typename.
+// So, in all mutations we have a return type of Promise<Order> and we are also casting the output i.e:
+// return response?.data?.X as Order
+
+const params: UseCartFactoryParams<Order, OrderLine, AgnosticProductVariant, Coupon> = {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  load: async (context: Context, { customQuery }) => {
-    console.log('Mocked: useCart.load');
-    return {};
+  load: async (context: Context, { customQuery }): Promise<Order> => {
+    const response = await context.$vendure.api.getCart(customQuery);
+
+    return response?.data?.activeOrder;
   },
 
+  // TODO: update to use currentCart
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  addItem: async (context: Context, { currentCart, product, quantity, customQuery }) => {
-    console.log('Mocked: useCart.addItem');
-    return {};
+  addItem: async (context: Context, { currentCart, product, quantity, customQuery }): Promise<Order> => {
+    const productVariantId = product?._variantId || product?._id;
+    const response = await context.$vendure.api.addToCart({ productVariantId, quantity }, customQuery);
+
+    return response?.data?.addItemToOrder as Order;
   },
 
+  // TODO: update to use currentCart
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  removeItem: async (context: Context, { currentCart, product, customQuery }) => {
-    console.log('Mocked: useCart.removeItem');
-    return {};
+  removeItem: async (context: Context, { currentCart, product, customQuery }): Promise<Order> => {
+    const response = await context.$vendure.api.removeFromCart({ orderLineId: product.id}, customQuery);
+
+    return response?.data?.removeOrderLine as Order;
   },
 
+  // TODO: update to use currentCart
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  updateItemQty: async (context: Context, { currentCart, product, quantity, customQuery }) => {
-    console.log('Mocked: useCart.updateItemQty');
-    return {};
+  updateItemQty: async (context: Context, { currentCart, product, quantity, customQuery }): Promise<Order> => {
+    const response = await context.$vendure.api.updateCartQuantity({ orderLineId: product.id, quantity }, customQuery);
+
+    return response?.data?.adjustOrderLine as Order;
   },
 
+  // Not used in VSF for now
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  clear: async (context: Context, { currentCart }) => {
+  clear: async (context: Context, { currentCart }): Promise<Order> => {
     console.log('Mocked: useCart.clear');
-    return {};
+    return Promise.resolve({}) as Promise<Order>;
```

Comment:
Don't we have to remove cart identifier from localStorage/cookie?

Reply:
This method was not implemented yet. It was not used in the template so I decided to leave these methods to develop later.